src/openfl/_internal/renderer/opengl/batcher/BatchRenderer.hx

```diff
@@ -129,7 +129,18 @@ class BatchRenderer {
 
 		// if the texture has changed - we need to either pack it into the current render group or create the next one
 		// and since on the first iteration the `currentTexture` is null, it's always "changed"
-		if (currentTexture != nextTexture) {
+		var textureChanged = currentTexture != nextTexture;
+
+		// if the texture was used in the current group (ticks are equal), but the smoothing mode has changed
+		// we gotta break the batch, because we can't render the same texture with different smoothing in a single batch
+		// TODO: we can in WebGL2 using Sampler objects
+		if (nextTexture.enabledTick == tick && nextTexture.lastSmoothing != quad.smoothing) {
+			finishCurrentGroup();
+			startNextGroup();
+			textureChanged = true; // act like the texture has changed
+		}
+
+		if (textureChanged) {
```

Comment:
According to the `blendMode` batch breaking, just do:

```
		if (nextTexture.enabledTick == tick && nextTexture.lastSmoothing != quad.smoothing) {
			currentTexture = null;

			finishCurrentGroup();
			startNextGroup();
		}

		if (currentTexture != nextTexture) {
			// ...
```

Reply:
Hm, looks good! Will do!

Reply:
Updated the PR